tests/test_connection.py

```diff
@@ -334,6 +391,132 @@ class MemoryListener(trio.abc.Listener):
         await client.send_message('Hello from client!')
 
 
+@fail_after(MAX_TIMEOUT_TEST_DURATION)
```

Comment:
isn't there some pytest setting to globally control test timeouts?

Reply:
I couldn't find one. There's [a plugin](https://pypi.org/project/pytest-timeout/) but I thought it would be nice to have Trio support for Trio's clock.